src/network/serverpackethandler.cpp

```diff
@@ -925,8 +925,8 @@ void Server::handleCommand_InventoryAction(NetworkPacket* pkt)
 
 		ma->from_inv.applyCurrentPlayer(player->getName());
 		ma->to_inv.applyCurrentPlayer(player->getName());
-
-		setInventoryModified(ma->from_inv, false);
+		if (ma->from_inv != ma->to_inv)
```

Comment:
Wow, we actually [have implemented](https://github.com/minetest/minetest/blob/7bbb9b066a8cc079512ddd3e6b32475309f49fca/src/inventorymanager.h#L69) this comparator.